client/ayon_core/tools/publisher/window.py

```diff
@@ -823,15 +830,44 @@ def _set_publish_comment(self):
         self._controller.set_comment(self._comment_input.text())
 
     def _on_validate_clicked(self):
-        if self._save_changes(False):
+        if self._start_publish_preflight() and self._save_changes(False):
             self._set_publish_comment()
             self._controller.validate()
 
     def _on_publish_clicked(self):
-        if self._save_changes(False):
+        if self._start_publish_preflight() and self._save_changes(False):
             self._set_publish_comment()
             self._controller.publish()
 
+    def _start_publish_preflight(self) -> bool:
+        # Validate comment length
+        comment_def = self._controller.get_comment_def()
+        if comment_def.minimum_chars_required:
+            char_count = len(self._comment_input.text())
+            if char_count < comment_def.minimum_chars_required:
+                self._overlay_object.add_message(
+                    "Please enter a comment of at least "
+                    f"{comment_def.minimum_chars_required} characters",
+                    message_type="error"
+                )
+                self._invalidate_comment_field()
+                return False
+        return True
```

Comment:
No need to optimise str lenght calculation.
```suggestion
        char_count = len(self._comment_input.text())
        if (
            comment_def.minimum_chars_required
            and char_count < comment_def.minimum_chars_required
        ):
            self._overlay_object.add_message(
                "Please enter a comment of at least "
                f"{comment_def.minimum_chars_required} characters",
                message_type="error"
            )
            self._invalidate_comment_field()
            return False
        return True
```